iroh-gateway/src/handlers.rs

```diff
@@ -550,16 +572,57 @@ async fn serve_fs_dir<T: ContentLoader + std::marker::Unpin>(
     if !root_path.ends_with('/') {
         root_path.push('/');
     }
+
+    let mut breadcrumbs: Vec<HashMap<&str, String>> = Vec::new();
+    root_path
+        .trim_matches('/')
+        .split('/')
+        .fold(&mut breadcrumbs, |accum, path_el| {
+            let mut el = HashMap::new();
+            let path = match accum.last() {
+                Some(prev) => {
+                    let base = prev.get("path");
+                    format!("/{}/{}", base.unwrap_or(&"".to_string()), encode(path_el))
```

Comment:
this will not do what you want, if `base` is empty this will produce `//<path_el>`.

```rust
match prev.get("path") {
  Some(base) => format!("/{}/{}", base, encode(path_el)),
  None => format!("/{}", encode(path_el)),
}
```
should be closer to what you want

Reply:
actually looking at the whole structure this can be simplified to 

```rust
match accum.last().and_then(|prev| prev.get("path")) {
    Some(base) => format!("/{}/{}", base, encode(path_el)),
    None => format!("/{}", encode(path_el)),
}
```